File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/orders/entity/Order.java
package com.portfolio.auctionmarket.domain.orders.entity;

import com.portfolio.auctionmarket.domain.auctions.entity.Auction;
import com.portfolio.auctionmarket.domain.products.entity.Product;
import com.portfolio.auctionmarket.domain.user.entity.User;
import com.portfolio.auctionmarket.global.base.BaseCreatedAt;
import jakarta.persistence.*;
import lombok.*;

@Getter
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
@Table(name = "orders", indexes = {
        @Index(name = "idx_order_buyer", columnList = "buyer_id"),
        @Index(name = "idx_order_auction", columnList = "auction_id")
})
@Builder
public class Order extends BaseCreatedAt {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id")
    private Long orderId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "auction_id")
    private Auction auction;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "buyer_id")
    private User buyer;

    @Column(name = "final_price")
    private Long finalPrice;

    public User getSeller() {
        return auction.getProduct().getSeller();
    }

    public String getTitle() {
        return auction.getProduct().getTitle();
    }

}

File: AuctionMarket/auction-domain/src/main/kotlin/org/example/auctioncommon/domain/order/entity/Order.kt
package org.example.auctioncommon.domain.order.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Index
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToOne
import jakarta.persistence.Table
import org.example.auctioncommon.domain.auction.entity.Auction
import org.example.auctioncommon.domain.product.entity.ProductStatus
import org.example.auctioncommon.domain.user.entity.User
import org.example.auctioncommon.global.base.BaseCreatedAt

@Entity
@Table(
    name = "orders",
    indexes = [Index(name = "idx_order_buyer", columnList = "buyer_id"), Index(
        name = "idx_order_auction",
        columnList = "auction_id"
    )]
)
class Order(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id")
    var orderId: Long? = null,

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "auction_id")
    val auction: Auction,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "buyer_id")
    val buyer: User,

    @Column(name = "final_price")
    var finalPrice: Long? = 0L,

) : BaseCreatedAt() {
    val seller: User?
        get() = auction.product?.seller

    val title: String?
        get() = auction.product?.title
}
